Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM python
+
+RUN ls /
+
```

Comment:
Actually, I'd recommend separating the dockerfile and the refactoring into separate PRs -- not only because they're unrelated, but also because there are so many refactored files, and it would be nice to have them all in one self-contained PR.  Of course let me know if you disagree.

In practice I'd recommend deleting the dockerfile from this PR and opening another with it, and leaving this PR for the refactoring.

Reply:
Sure!